src/Fresnel/Profunctor/Optical.hs

```diff
@@ -0,0 +1,98 @@
+module Fresnel.Profunctor.Optical
+( IsIso
+, IsLens
+, IsGetter
+, IsPrism
+, IsReview
+, IsAffineTraversal
+, IsTraversal
+, IsAffineFold
+, IsFold
+, IsSetter
+) where
+
+
+import Data.Profunctor
+import Fresnel.Bifunctor.Contravariant
+import Fresnel.Profunctor.Recall
+import Fresnel.Profunctor.Coexp
+import Data.Bifunctor
+import Data.Profunctor.Traversing
+import Fresnel.Profunctor.AffineStar
+import Data.Functor.Contravariant
+import Control.Arrow
+
+class Profunctor p => IsIso p
+
+instance IsIso (->)
+instance Monad m => IsIso (Kleisli m)
+instance IsIso (Forget r)
+instance IsIso (Recall e)
+instance Functor f => IsIso (Star f)
+instance Functor f => IsIso (Costar f)
+instance Functor f => IsIso (AffineStar f)
+instance IsIso (Coexp s t)
```

Comment:
I’ve placed instances in this module wherever it wouldn’t require resolving cyclic imports so as to make it obvious which types can be used with which optic families.